libs/core/type_support/tests/unit/relocate_at.cpp

```diff
@@ -0,0 +1,133 @@
+//  Copyright (c) 2023 Isidoros Tsaousis-Seiras
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/init.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/type_support/construct_at.hpp>
+#include <hpx/type_support/relocate_at.hpp>
+
+struct non_trivially_relocatable_struct
+{
+    static int count;
+    int data;
+
+    non_trivially_relocatable_struct(int data)
+      : data(data)
+    {
+        count++;
+    }
+    non_trivially_relocatable_struct(non_trivially_relocatable_struct&& other)
```

Comment:
```suggestion
    non_trivially_relocatable_struct(non_trivially_relocatable_struct&& other) noexcept
```
The same change needs to be applied in other spots as well.

Reply:
One point of relocation is that it can bypass any exceptions that a move constructor would throw, since it is just a memcpy, so I think we should leave this with exceptions on